src/loops/simple_programs_with_repitition/easy/q8/MainTest.java

```diff
@@ -0,0 +1,63 @@
+package loops.simple_programs_with_repitition.easy.q8;
+
+import global.variables.*;
+import global.variables.clauses.*;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import java.util.stream.Stream;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import global.BaseTest;
+import global.exceptions.InvalidClauseException;
+import global.tools.TestOption;
+
+public class MainTest extends BaseTest {
+    // Parsons
+
+    public Clause[] testSentence() {
+        TestOption.isInputTest = true;
+        TestOption.defaultInput = "This is a sentence. This is another sentence." + System.lineSeparator() + "2";
+
+        return new Clause[] {
+            new StringLiteral("Enter a sentence: "),
+            new NewLine(),
+            new StringLiteral("Enter a number to offset by: "),
+            new NewLine(),
+            new StringLiteral("Your Encrypted Sentence is: "),
+            new PlaceHolder()
+        };
+        
+    }
+
+    public void runMain() {
+        CaesarSike.main(new String[0]);
+    }
+
+    public static String encrypt(String sentence, int offset) {
+        String result = "";
+        int i = 0;
+        while (i < sentence.length()) {
+            result += (char)(sentence.charAt(i) + offset);
+            i++;
+        }
+        return result;
+    }
+
+    static Stream<Arguments> inputProvider(){
+        return Stream.of(
+            Arguments.of("This is just for testing. Do not copy!", 1, encrypt("This is just for testing. Do not copy!", 1)),
+            Arguments.of("This is second test", 100, encrypt("This is second test", 100)),
+            Arguments.of("hard", 1, "ibse"),
+            Arguments.of(" ", 1, "!")
+        ); // Doesn't work with empty string
```

Comment:
Wait wouldn't empty string just return empty string regardless of offset? If it doesn't why not

Reply:
space is also considered a character in ASCII table, so putting offset on space also changes the character.